server/src/test/java/org/opensearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -528,6 +550,544 @@ public void onFailure(Exception e) {
+        assertEquals(numTasks * 2, fetchPhaseEnd.get());
+    }
+
+    private SearchDfsQueryThenFetchAsyncAction createSearchDfsQueryThenFetchAsyncAction() {
+        final TransportSearchAction.SearchTimeProvider timeProvider = new TransportSearchAction.SearchTimeProvider(
+            0,
+            System.nanoTime(),
+            System::nanoTime
+        );
+        SearchTransportService searchTransportService = new SearchTransportService(null, null);
+        Map<String, Transport.Connection> lookup = new ConcurrentHashMap<>();
+        SearchPhaseController controller = new SearchPhaseController(
+            writableRegistry(),
+            r -> InternalAggregationTestCase.emptyReduceContextBuilder()
+        );
+        final SearchRequest searchRequest = new SearchRequest();
+        searchRequest.allowPartialSearchResults(false);
+
+        int numConcurrent = randomIntBetween(1, 4);
+        searchRequest.setMaxConcurrentShardRequests(numConcurrent);
+        searchRequest.setBatchedReduceSize(2);
+        searchRequest.source(new SearchSourceBuilder().size(1).sort(SortBuilders.fieldSort("timestamp")));
+        SearchTask task = new SearchTask(0, "n/a", "n/a", () -> "test", null, Collections.emptyMap());
+        int numShards = 1;
+        Executor executor = OpenSearchExecutors.newDirectExecutorService();
+        DiscoveryNode primaryNode = new DiscoveryNode("node_1", buildNewFakeTransportAddress(), Version.CURRENT);
+        DiscoveryNode replicaNode = new DiscoveryNode("node_2", buildNewFakeTransportAddress(), Version.CURRENT);
+        GroupShardsIterator<SearchShardIterator> shardsIter = SearchAsyncActionTests.getShardsIter(
+            "idx",
+            new OriginalIndices(new String[] { "idx" }, SearchRequest.DEFAULT_INDICES_OPTIONS),
+            numShards,
+            randomBoolean(),
+            primaryNode,
+            replicaNode
+        );
+        QueryPhaseResultConsumer resultConsumer = new QueryPhaseResultConsumer(
+            searchRequest,
+            executor,
+            new NoopCircuitBreaker(CircuitBreaker.REQUEST),
+            controller,
+            task.getProgressListener(),
+            writableRegistry(),
+            shardsIter.size(),
+            exc -> {}
+        );
+        return new SearchDfsQueryThenFetchAsyncAction(
+            logger,
+            searchTransportService,
+            (clusterAlias, node) -> lookup.get(node),
+            Collections.singletonMap("_na_", new AliasFilter(null, Strings.EMPTY_ARRAY)),
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            controller,
+            executor,
+            resultConsumer,
+            searchRequest,
+            null,
+            shardsIter,
+            timeProvider,
+            null,
+            task,
+            SearchResponse.Clusters.EMPTY
+        );
+    }
+
+    private SearchQueryThenFetchAsyncAction createSearchQueryThenFetchAsyncAction() {
```

Comment:
This and below create functions are doing a lot here. Can't we just utilize existing createAction() inside this tests class? I will also take a look deeply.

Reply:
I took these changes from the SearchQueryThenFetchAsyncActionTests class and looked at how they initialized their actions.